File: server/recipe.go
package main

type VertexResponse struct {
	Recipes []VertexRecipe `json:"recipes"`
}

type VertexRecipe struct {
	Name             string             `json:"name"`
	Ingredients      []VertexIngredient `json:"ingredients"`
	Instructions     []string           `json:"instructions"`
	PrepTimeMinutes  int                `json:"prepTimeMinutes"`
	TotalTimeMinutes int                `json:"totalTimeMinutes"`
}

type VertexIngredient struct {
	Name        string `json:"name"`
	Quantity    string `json:"quantity"`
	Unit        string `json:"unit"`
	Preparation string `json:"preparation"`
}

File: client/src/model/recipe.ts
export interface Recipe {
  name: string;
  ingredients: Ingredient[];
  instructions: string[];
  prepTimeMinutes: number;
  totalTimeMinutes: number;
}

export interface Ingredient {
  name: string;
  quantity: string;
  unit: string;
  preparation: string;
}

export interface SmartImport {
  id: string;
  status: "success" | "processing" | "error";
  url: string;
  rawText: string;
  error: any;
  recipes: Recipe[];
}
